### DISCUSSION TITLE
How can I store the data directly into my own database?

### DISCUSSION THREAD

**Discussion by jiojiojackson** [ActivityWatch/activitywatch#1022] [Q&A] [answered]
How can I store the data directly into my own database? For example, MySql

**Comment by BelKed** [comment, marked as answer]
If you are familiar with Python, you can use the [`aw-client`](https://github.com/ActivityWatch/aw-client) library to retrieve the data from ActivityWatch. After processing the data in the Python script, you can use another library to communicate with your MySQL database to store the data there.

If you want to use any other programming language, you can connect to ActivityWatch via the [API](https://docs.activitywatch.net/en/latest/api/rest.html) and the process would be similar.

---

If you want to avoid using the ActivityWatch server altogether, you can listen on the default ActivityWatch port (`5600`) where all watchers send their data, and then store the data in your database.
